lib/stages/basebandReadout.cpp

```diff
@@ -175,167 +199,230 @@ void basebandReadout::main_thread() {
     }
 }
 
-void basebandReadout::readout_thread(const uint32_t freq_id, basebandReadoutManager& mgr) {
-    auto& request_no_data_counter = readout_counter.labels({std::to_string(freq_id), "no_data"});
+void basebandReadout::readout_thread(const uint32_t freq_ids[], basebandReadoutManager *mgrs[]) {
+    //uint32_t freq_id = freq_ids[_num_local_freq - 1];
+    //basebandReadoutManager& mgr = *(mgrs[_num_local_freq - 1]);
+    std::unique_ptr<basebandReadoutManager::requestStatusMutex> next_requests[_num_local_freq];
+    std::shared_ptr<basebandReadoutManager::requestStatusMutex> next_request;
+    kotekan::prometheus::Counter *request_no_data_counters[_num_local_freq]; 
+    for(int freqidx = 0; freqidx < _num_local_freq; freqidx++){
+        request_no_data_counters[freqidx] = &(readout_counter.labels({std::to_string(freq_ids[freqidx]), "no_data"}));
+    }
+
+    //TODO: Ask davor about request_no_data_counter()
 
     while (!stop_thread) {
         // Code that listens and waits for triggers and fills in trigger parameters.
         // Latency is *key* here. We want to call get_data within 100ms
         // of L4 sending the trigger.
+        next_requests[0] = mgrs[0]->get_next_waiting_request();
```

Comment:
Where is `next_requests[]` declared?

Reply:
Line 201, it looks like an array of `std::unique_ptr<basebandReadoutManager::requestStatusMutex>`. Why is it a pointer to mutex?

Reply:
Just because there's no default constructor for a mutex. I need the mutexes to not go out of scope over the entire readout_thread(), right?